code/game/objects/items/devices/illegalradio.dm

```diff
@@ -194,22 +233,26 @@
 	dat = jointext(dat,"") 
 	return dat	
 
+
+	
+	
 /obj/item/device/illegalradio/proc/dispense_change()
 	if(money_stored > 0)
 		dispense_cash(money_stored,get_turf(src))
 		money_stored = 0
 	interact(usr)
 
 /obj/item/device/illegalradio/afterattack(atom/A as mob|obj, mob/user as mob)
-	if(istype(A, /obj/item/weapon/spacecash))
+	if(istype(A, /obj/item/weapon/spacecash) && A.Adjacent(user))
```

Comment:
Why not (pseudocode):
```
if(adjacent)
  if(istype(A, relevant/thing)
    do stuff
  else if other relevant thing
   do other stuff
else ..()
```

Reply:
The difference in speed is in the ten-thousandths of seconds for a function executed quite rarely, so I think it's a fine sacrifice for more organized code.